lib/Doctrine/DBAL/Driver/Mysqli/MysqliConnection.php

```diff
@@ -239,7 +239,7 @@ private function setDriverOptions(array $driverOptions = [])
         $exceptionMsg = "%s option '%s' with value '%s'";
 
         foreach ($driverOptions as $option => $value) {
-            if ($option === static::OPTION_FLAGS) {
+            if ($option === self::OPTION_FLAGS) {
```

Comment:
Wondering if this is BC compliant: does any subclass override this?

Reply:
Even if we don't some dependencies in theory may. I'd disable the rule for `master` and add it in `develop`. In order to avoid further confusion, all affected constants (not sure what specifically the rule is above) will need to be marked `private`~/`final`~ …we don't have `final const` in PHP?!

Reply:
I did not see any subclass for MysqliConnection,                 
I tried to find `final const` but I did not find in any, https://3v4l.org/NU0ic#output         

The rule of prohibiting late static binding is only for constants, feels logical, late static binding is always annoying         
SlevomatCodingStandard.Classes.DisallowLateStaticBindingForConstants.DisallowedLateStaticBindingForConstant

but I'm sure you have seen much more cases than me :D , so what should be done ?

Reply:
@morozov's plan makes sense. This change is a potential BC break for downstream consumers, so we can only apply it for `develop` (which is for the next major release)

Reply:
Should I re-base this PR against the develop ?

Reply:
@amaabdou please revert all `static` → `self` changes from this PR and keep it targeting master. Since the coding standard will report the violations, please exclude the corresponding rules for those files.

Once this is merged, I'll rebase `develop` and we can apply the breaking changes.